### PR TITLE
Group chat packets

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/groupchat_packets.md [hunk 1]
```diff
@@ -0,0 +1,1128 @@
...
+- if the group doesn't exist
+    - return error.
+- else if the group status is not `connected`
+    - return error.
+- else
+    - assemble and send `Invite packet` using messenger and net crypto.
+    
+When a peer receives `Invite packet` then:
+
+- check if the packet length is correct, if it is not correct
```
Comment (kpp) [inline src/groupchat_packets.md]: This is too detailed

File: src/groupchat_packets.md [hunk 2]
```diff
@@ -0,0 +1,1128 @@
...
+
+When a peer receives this packet then:
+
+- kill the friend connection.
+- set close list to nothing connected.
+
+## Peer query packet
+
+Sent to peer to query the peer list in a group chat. A peer receives this packet send response packet with `response packet`.
+If there are many peers in a group chat, then response packet may larger than the maximum packet size of friend connection packet(1373 bytes),
```
Comment (kpp) [inline src/groupchat_packets.md]: > packet may larger

may be larger

File: src/groupchat_packets.md [hunk 3]
```diff
@@ -0,0 +1,1128 @@
...
+then multiple response packets will be sent.
+
+#### Request
```
Comment (kpp) [inline src/groupchat_packets.md:121]: > Request | Response

Does it mean we have 2 different packets under the same id?

File: src/groupchat_packets.md [hunk 4]
```diff
@@ -0,0 +1,1128 @@
...
+Sent to send action to all member of group chat.
+
+When a peer receives this packet then:
+
+- show the message in the chatting window by calling callback of client.
+
+
+# DHT based group chat(version 2)
+
+New version of group chat is based on DHT.
```
Comment (kpp) [inline src/groupchat_packets.md:335]: Are there any docs from toktok team? Can you add link to them?

File: src/groupchat_packets.md [hunk 5]
```diff
@@ -0,0 +1,1128 @@
...
+    - return error.
+- call callback of client to update UI.
+
+#### Sync Request (0xf8)
+
+Serialized form:
+
+Length      | Content
+------------|-------
+`32`        | `password`(comes only when password is setted)
```
Comment (kpp) [inline src/groupchat_packets.md:710]: > password

plaintext?

**Comment by kpp** [comment]
good job!

File: src/groupchat_packets.md (hunk 3 above)
Comment (NamsooCho) [reply src/groupchat_packets.md:121]: Yes.
But 4th bytes are different.
```
#### Request

Serialized form:

Length    | Content
--------- | ------
`1`       | `0x62`
`2`       | `group number`
`1`       | `0x08`
```

```
Length    | Content
--------- | ------
`1`       | `0x62`
`2`       | `group number`
`1`       | `0x09`
```

File: src/groupchat_packets.md (hunk 5 above)
Comment (NamsooCho) [reply src/groupchat_packets.md:710]: Yes.
It was encrypted with shared secret as packet's payload and there is no additional encryption.

**Pull request merged**